src/main/scala/edu/berkeley/cs/rise/opaque/Utils.scala

```diff
@@ -293,18 +279,94 @@ object Utils extends Logging {
 
   var eid = 0L
   var attested: Boolean = false
-  var attesting_getepid: Boolean = false
-  var attesting_getmsg1: Boolean = false
-  var attesting_getmsg3: Boolean = false
-  var attesting_final_ra: Boolean = false
+  // Initialize accumulator variables for tracking the state of attestation
+  var acc_registered: Boolean = false
+  val numUnattested: LongAccumulator = new LongAccumulator
+  val numAttested: LongAccumulator = new LongAccumulator
 
   def initSQLContext(sqlContext: SQLContext): Unit = {
     sqlContext.experimental.extraOptimizations =
       (Seq(EncryptLocalRelation, ConvertToOpaqueOperators) ++
         sqlContext.experimental.extraOptimizations)
     sqlContext.experimental.extraStrategies = (Seq(OpaqueOperators) ++
       sqlContext.experimental.extraStrategies)
-    RA.initRA(sqlContext.sparkContext)
+
+    val sc = sqlContext.sparkContext
+    if (!acc_registered) {
```

Comment:
This is to prevent the Utils object from reused across multiple tests, and causing an error because the accumulator variables are re-registered. I added a comment about this.